test/common/http/utility_test.cc

```diff
@@ -1477,5 +1477,24 @@ TEST(PercentEncoding, Encoding) {
   EXPECT_EQ(Utility::PercentEncoding::encode("too%!large/", "%!/"), "too%25%21large%2F");
 }
 
+TEST(CheckRequiredRequestHeaders, OK) {
+  EXPECT_EQ(Http::okStatus(), HeaderUtility::checkRequiredRequestHeaders(
+                                  TestRequestHeaderMapImpl{{":method", "GET"}, {":path", "/"}}));
+  EXPECT_EQ(Http::okStatus(), HeaderUtility::checkRequiredRequestHeaders(TestRequestHeaderMapImpl{
+                                  {":method", "CONNECT"}, {":authority", "localhost:1234"}}));
+}
+
+TEST(CheckRequiredRequestHeaders, NG) {
```

Comment:
add response header tests?
(very minor nit, but i'm not sure what NG stands for, can you make this slightly more verbose?)

Reply:
https://github.com/envoyproxy/envoy/pull/16745/commits/7953e305afe2be0d312223b87ab4025e18927965 done, thanks!